crates/miden-tx/src/prover/prover_host.rs

```diff
@@ -51,13 +51,15 @@ where
         mast_store: &'store STORE,
         scripts_mast_store: ScriptMastForestStore,
         acct_procedure_index_map: AccountProcedureIndexMap,
+        source_manager: Arc<dyn SourceManagerSync>,
```

Comment:
Maybe there was a discussion I missed, but I think the prover host doesn't need to deal with the source manager, since we can usually assume that once the prover is called, execution has previously succeeded, and so we should not run into errors. In particular, because the source manager is only used to improve error messages, but is not necessary for overall correctness.

So I think I'd move from the base host into the `TransactionExecutorHost`.

Reply:
Fixed